### PR TITLE
Basic tests to catch edge cases

### PR DESCRIPTION
**Pull request by odow**
- Relating to ScalarQuadraticFunctions:
(1) duplicate terms on the diagonal; and
(2) duplicate terms on the off-diagonal, including duplicates of elements at locations (i,j) and (j,i).
- Deleting mixtures of scalaraffine and vectoraffine functions.

### CODE REVIEW

File: src/Test/UnitTests/constraints.jl [hunk 1]
```diff
@@ -132,3 +132,119 @@ function solve_affine_interval(model::MOI.ModelLike, config::TestConfig)
...
+"""
+function solve_qcp_edge_cases(model::MOI.ModelLike, config::TestConfig)
+    @testset "Duplicate on-diagonal" begin
+        # max x + 2y | y + x^2 + x^2 <= 1, x >= 0.5, y >= 0.5
+        MOI.empty!(model)
+        x = MOI.addvariables!(model, 2)
+        MOI.set!(model, MOI.ObjectiveSense(), MOI.MaxSense)
+        MOI.set!(model,
+            MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
+            MOI.ScalarAffineFunction{Float64}(
```
Comment (blegat) [inline src/Test/UnitTests/constraints.jl]: You don't have to explicitely specify the type `{Float64}`. At least it is inconsistent with other tests
